### ISSUE TITLE
When i merge namespaces with classes ,typescript reminded me to no-redeclare?

### ISSUE THREAD

**Issue by ttttttssssss** [typescript-eslint/typescript-eslint#3864]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```js
module.exports = {
  env: {
    browser: true,
    es2021: true
  },
  extends: [
    'standard'
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module'
  },
  plugins: [
    '@typescript-eslint'
  ],
  rules: {
  }
}

```

```TS
// your repro code case
class BST {
    root:BST.Node
    constructor (key:number, value:any) {
      this.root = new BST.Node(key, value)
    }
}
// error: 'BST' is already defined.
namespace BST{
    export class Node {
        key:number
        value:any
        constructor (key:number, value:any) {
          this.key = key
          this.value = value
        }
    }
}
export {
  BST
}

```

**Expected Result**
No Error

**Actual Result**
'BST' is already defined. eslint(no-redeclare) [8,11]

**Additional Info**
I'm codeing from the documents, Why is this a problem?
document: https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-namespaces

Is there an alternative way to write it？

Another example
```ts
function Foo () {}
// 'Foo' is already defined.
namespace Foo{
    export const a = 10
}
export {
  Foo
}

```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.3` |
| `@typescript-eslint/parser`        | `4.29.3` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.3` |

**Comment by bradzacher** [comment]
You ticked the "I have read the FAQ and my problem is not listed" box... yet the first FAQ answer tells you the answer.

https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code
